packages/opencensus-core/src/trace/instrumentation/base-plugin.ts

```diff
@@ -60,10 +78,59 @@ export abstract class BasePlugin implements types.Plugin {
   // while unpatching is not. It should be the other way around
 
   // tslint:disable:no-any
-  applyPatch(moduleExports: any, tracer: modelTypes.Tracer, version: string):
-      any {
-    this.setPluginContext(moduleExports, tracer, version);
+  applyPatch(
+      moduleExports: any, tracer: modelTypes.Tracer, version: string,
+      basedir?: string): any {
+    this.setPluginContext(moduleExports, tracer, version, basedir);
   }
 
   abstract applyUnpatch(): void;
+
+
+  /**
+   * Load internal files according to version range
+   */
+  private loadInternalFiles(): types.ModuleExports {
+    let result: types.ModuleExports = null;
+    this.logger.debug('loadInternalFiles %o', this.internalFileList);
+    if (this.internalFileList) {
+      Object.keys(this.internalFileList).map(versionRange => {
+        if (semver.satisfies(this.version, versionRange)) {
+          result = this.loadInternalModuleFiles(
```

Comment:
Should you warn if multiple versions match? It sounds like users would be depending on implementation-private behavior.

Reply:
If multiple versions match, the last one will prevail.

Reply:
Can you add a logger warning? It should be brought to the user's attention because it is mostly a bug for someone to user overlapping version ranges.